File: be/Models/AllCodes.cs
namespace be.Models
{
    public class AllCodes
    {
        public int CodeID { get; set; } // Primary key, INTEGER, auto-increment
        public string Type { get; set; } // STRING(30), not null
        public string Code { get; set; } // STRING(20), not null
        public string CodeValueVI { get; set; } // STRING(50), not null
        public decimal? ExtraValue { get; set; } // DECIMAL(10,2), nullable
    }
}

File: be/src/models/AllCodes.js
'use strict';
const { Model } = require('sequelize');

module.exports = (sequelize, DataTypes) => {
  class AllCodes extends Model {
    static associate(models) {
      // Không có mối quan hệ trực tiếp, nhưng được tham chiếu bởi nhiều bảng
    }
  }

  AllCodes.init(
    {
      CodeID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
        allowNull: false,
      },
      Type: {
        type: DataTypes.STRING(30),
        allowNull: false,
      },
      Code: {
        type: DataTypes.STRING(20),
        allowNull: false,
      },
      CodeValueVI: {
        type: DataTypes.STRING(50),
        allowNull: false,
      },
      ExtraValue: {
        type: DataTypes.DECIMAL(10, 2),
        allowNull: true,
      },
    },
    {
      sequelize,
      modelName: 'AllCodes',
      tableName: 'AllCodes',
      timestamps: false,
      indexes: [
        { unique: true, fields: ['Type', 'Code'], name: 'unique_type_code' },
        { fields: ['Type'], name: 'index_type' },
      ],
    }
  );

  return AllCodes;
};